nni/mutable/symbol.py

```diff
@@ -0,0 +1,472 @@
+        return self.repr_template.format(*reprs)
+
+    # Python special methods list:
+    # https://docs.python.org/3/reference/datamodel.html#special-method-names
+
+    # Special operators that can be useful in place of built-in conditional operators.
+    @symbolic_staticmethod
+    def to_int(obj: Any) -> SymbolicExpression | int:
+        """Convert the current value to an integer."""
+        if isinstance(obj, SymbolicExpression):
+            return obj.expr_cls(int, 'int({})', [obj])
+        return int(obj)
+
+    @symbolic_staticmethod
+    def to_float(obj: Any) -> SymbolicExpression | float:
+        """Convert the current value to a float."""
+        if isinstance(obj, SymbolicExpression):
+            return obj.expr_cls(float, 'float({})', [obj])
+        return float(obj)
+
+    @symbolic_staticmethod
+    def condition(pred: Any, true: Any, false: Any) -> SymbolicExpression | Any:
+        """
+        Return ``true`` if the predicate ``pred`` is true else ``false``.
+
+        Examples
+        --------
+        >>> SymbolicExpression.condition(Symbol('x') > Symbol('y'), 2, 1)
+        """
+        symbol_obj = first_symbolic_object(pred, true, false)
+        if symbol_obj is not None:
+            return symbol_obj.expr_cls(lambda t, c, f: t if c else f, '{} if {} else {}', [true, pred, false])
+        return true if pred else false
+
+    @symbolic_staticmethod
+    def max(arg0: Iterable[Any] | Any, *args: Any) -> Any:
+        """
+        Returns the maximum value from a list of symbols.
+        The usage should be similar to Python's built-in symbols,
+        where the parameters could be an iterable, or at least two arguments.
+        """
+        if not args:
+            if not isinstance(arg0, Iterable):
+                raise TypeError('Expect more than one items to compare max')
+            return SymbolicExpression.max(*list(arg0))
+        lst = list(arg0) if isinstance(arg0, Iterable) else [arg0] + list(args)
+        symbol_obj = first_symbolic_object(*lst)
+        if symbol_obj is not None:
+            return symbol_obj.expr_cls(max, 'max({})', lst)
+        return max(lst)
+
+    @symbolic_staticmethod
+    def min(arg0: Iterable[Any] | Any, *args: Any) -> Any:
+        """
+        Returns the minimum value from a list of symbols.
+        The usage should be similar to Python's built-in symbols,
+        where the parameters could be an iterable, or at least two arguments.
+        """
+        if not args:
+            if not isinstance(arg0, Iterable):
+                raise TypeError('Expect more than one items to compare min')
+            return SymbolicExpression.min(*list(arg0))
+        lst = list(arg0) if isinstance(arg0, Iterable) else [arg0] + list(args)
+        symbol_obj = first_symbolic_object(*lst)
+        if symbol_obj is not None:
+            return symbol_obj.expr_cls(min, 'min({})', lst)
+        return min(lst)
+
+    def __hash__(self):
```

Comment:
why required?

Reply:
Couldn't really recall. I guess it's because hash won't work because `__eq__` is overridden.

Reply:
suggestion: if hash isn't needed, '__hash__' can raise an error. '__eq__' will not return a bool so the hash cannot use.
but if hash is needed, we can determine whether to return a bool by detecting calling frame in '__hash__' and '__eq__'

Reply:
I don't like calling frame because they are notoriously nasty and I don't think it's worth the effort here.

Left unimplemented can be an option. I just deleted `__hash__`. I guess I can add it back if some tests complain in future.